### PR TITLE
add cookieRefresh transparency to GVL

### PR DESCRIPTION
**Pull request by alextcone**
TCF v2 Policy 16(2bis) currently prohibits Vendors from refreshing the maximum storage duration without re-consent from the user. Re-consent may never happen (or at least not within a useful timeframe) since the Policy only requires users to be reminded of their choices every 13 months. The aim of this update is to remove this prohibition and instead introduce transparency via an additional field in the GVL for vendors to indicate whether refresh can happen.
 
This tech spec already requires Vendors to include the initial lifespan of the cookie without accounting for duration extensions where storage is refreshed so this would not mean major changes for Vendor registration. The spec does, however, need to accommodate an additional Global Vendor List Field below the cookieMaxAgeSeconds field (in addition to granular disclosures in deviceStorage.json) so that Vendors can provide transparency on initial duration and, in addition, on whether such duration is susceptible to refresh.

### CODE REVIEW

**Review by janwinkler — APPROVED** [review]

**Review by dmdabbs — APPROVED** [review]

**Review by dmdabbs — APPROVED** [review]

**Review by dmdabbs — APPROVED** [review]

File: TCFv2/IAB Tech Lab - Device storage duration and access disclosure.md [hunk 1]
```diff
@@ -179,6 +210,16 @@ The deviceStorage.json is hosted at the vendor-supplied URL (<code>deviceStorage
...
    </td>
   </tr>
+  <tr>
+   <td><code>cookieRefresh</code>
+   </td>
+   <td><strong>optional</strong>
+   </td>
+   <td>boolean
+   </td>
+   <td>Only required for the type = ‘cookie’; otherwise false. Indicates the vendor’s refreshing a cookie (see example above). True indicates the vendor refreshes this cookie. False indicates the vendor does not refresh the this cookie any time the browser reloads.
```
Comment (dmdabbs) [inline TCFv2/IAB Tech Lab - Device storage duration and access disclosure.md]: Delete " this"

**Review by dmdabbs — APPROVED** [review]

**Comment by dmdabbs** [comment]
Howdy @alextcone 
I made the following changes while reviewing:
* Removed extra word, possessive change and adding cookieRefresh to the example.
* And some other apparently changeless commit.
* And I reviewed and approved the preceding three commits.

File: TCFv2/IAB Tech Lab - Device storage duration and access disclosure.md [hunk 2]
```diff
@@ -47,6 +47,37 @@ The number of seconds representing the longest potential duration for cookie sto
   </tr>
 </table>
 
+### <code>cookieRefresh</code>
+
+This true or false field indicates whether the cookie in scope for <code>cookieMaxAgeSeconds</code> is refreshed when a browser reloads. The following is an example of a cookie "refresh" scenario. On Day 0 a user visits a webpage which loads Vendor A who seeks Purpose 1 consent. She consents to Purpose 1 which includes Vendor A's 90 day max age cookie disclosure. This cookie is set to expire 90 days from now on Day 90 using <code>Set-Cookie: Max-Age=7776000</code>. On Day 5 the same user again visits the same webpage loading Vendor A. The webpage's CMP previously recorded her data processing choices and does not do so again. Vendor A is considered to "refresh" the cookie if it resets the countdown to 90 days from day 5, which would now be 96 days after the user made her choice when the webpage's CMP displayed a transparency and control experience to her.
```
Comment (Facens) [inline TCFv2/IAB Tech Lab - Device storage duration and access disclosure.md:52]:
```suggestion
This true or false field indicates whether any cookies in scope for <code>cookieMaxAgeSeconds</code> are refreshed after being initially set. The following is an example of a cookie "refresh" scenario. On Day 0 a user visits a webpage which loads Vendor A who seeks Purpose 1 consent. She consents to Purpose 1 which includes Vendor A's 90 day max age cookie disclosure. This cookie is set to expire 90 days from now on Day 90 using <code>Set-Cookie: Max-Age=7776000</code>. On Day 5 the same user again visits the same webpage loading Vendor A. The webpage's CMP previously recorded her data processing choices and does not surface a new consent request. Vendor A is considered to "refresh" the cookie if it resets the countdown to 90 days from day 5, which would now be 96 days after the user made her choice when the webpage's CMP displayed a transparency and control experience to her.
```

**Review by Facens — APPROVED** [review]
Made some minor suggestions to improve clarity, feel free to consider them, but approving.

**Comment by alextcone** [comment]
This is ready to merge at the request of IAB Europe TCF MO via the Policies Working Group and Steering Group. Actual GVL updates are managed by IAB Europe (not IAB Tech Lab) and will be reflected in upcoming weekly updates to the GVL.

**Pull request merged**